File: Q28 Valid parenthesis.js
var minSwaps = function(nums) {
    let k = nums.reduce((acc, num) => acc + num, 0);
    let n = nums.length;
    let cnt = nums.slice(0, k).reduce((acc, num) => acc + num, 0);
    let mx = cnt;
    for (let i = k; i < n + k; ++i) {
        cnt += nums[i % n] - nums[(i - k + n) % n];
        mx = Math.max(mx, cnt);
    }
    return k - mx;
};

// Example usage
console.log(minSwaps([0, 1, 0, 1, 1, 0, 0]));  // Output: 1

File: Q28 Valid parenthesis.py
class Solution:
    def minSwaps(self, nums):
        k = sum(nums)
        n = len(nums)
        cnt = sum(nums[:k])
        mx = cnt
        for i in range(k, n + k):
            cnt += nums[i % n] - nums[(i - k + n) % n]
            mx = max(mx, cnt)
        return k - mx

# Example usage
solution = Solution()
print(solution.minSwaps([0, 1, 0, 1, 1, 0, 0]))  # Output: 1
